### PR TITLE
Document behavior of areas with concave shapes

### PR DESCRIPTION
**Pull request by rburing**
Fixes the documentation part of https://github.com/godotengine/godot/issues/64588.

### CODE REVIEW

File: doc/classes/Area2D.xml [hunk 1]
```diff
@@ -4,7 +4,9 @@
 		2D area for detection and physics and audio influence.
 	</brief_description>
 	<description>
-		2D area that detects [CollisionObject2D] nodes overlapping, entering, or exiting. Can also alter or override local physics parameters (gravity, damping) and route audio to a custom audio bus.
+		2D area that detects [CollisionObject2D] nodes overlapping, entering, or exiting. Can also alter or override local physics parameters (gravity, damping) and route audio to custom audio buses.
+		To give the area its shape, add a [CollisionShape2D] or a [CollisionPolygon2D] node as a [i]direct[/i] child (or add multiple such nodes as direct children) of the area.
+		[b]Warning[/b]: See [ConcavePolygonShape2D] for a warning about possible unexpected behavior when using such a shape for an area.
```
Comment (akien-mga) [inline doc/classes/Area2D.xml:9]:
```suggestion
		[b]Warning:[/b] See [ConcavePolygonShape2D] for a warning about possible unexpected behavior when using such a shape for an area.
```

File: doc/classes/Area3D.xml [hunk 2]
```diff
@@ -5,6 +5,8 @@
 	</brief_description>
 	<description>
 		3D area that detects [CollisionObject3D] nodes overlapping, entering, or exiting. Can also alter or override local physics parameters (gravity, damping) and route audio to custom audio buses.
+		To give the area its shape, add a [CollisionShape3D] or a [CollisionPolygon3D] node as a [i]direct[/i] child (or add multiple such nodes as direct children) of the area.
+		[b]Warning[/b]: See [ConcavePolygonShape3D] (also called "trimesh") for a warning about possible unexpected behavior when using such a shape for an area.
```
Comment (akien-mga) [inline doc/classes/Area3D.xml:9]:
```suggestion
		[b]Warning:[/b] See [ConcavePolygonShape3D] (also called "trimesh") for a warning about possible unexpected behavior when using such a shape for an area.
```

**Review by fabriceci — APPROVED** [review]
LGTM !

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks, this is helpful :) Yay for better docs!

**Comment by akien-mga** [comment]
This might be relevant to backport to `3.x`, though it doesn't seem too trivial as some nodes were renamed, the documentation for some has changed in `master` and CollisionPolygonShape doesn't have `backface_collision`.

**Pull request merged**